src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/attachment/v2/AwsV2AttachmentPlugin.java

```diff
@@ -59,87 +55,87 @@ public boolean hasFailed(String instanceState) {
-		LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE, attachmentOrder.getInstanceId(), 
-				cloudUser.getToken()));
-
-		String volumeId = attachmentOrder.getVolumeId();
-		DetachVolumeRequest request = DetachVolumeRequest.builder()
-				.volumeId(volumeId)
-				.build();
-
-		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
-		try {
-			client.detachVolume(request);
-		} catch (Ec2Exception e) {
-			LOGGER.error(String.format(Messages.Error.ERROR_WHILE_REMOVING_RESOURCE, RESOURCE_NAME,
-					attachmentOrder.getInstanceId()), e);
-
-			throw new UnexpectedException();
-		}
+		return doAttachVolumeRequest(client, request);
 	}
 
-	@Override
-	public AttachmentInstance getInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
-		LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE, attachmentOrder.getInstanceId(), 
-				cloudUser.getToken()));
-
-		String attachmentId = attachmentOrder.getInstanceId();
-		DescribeVolumesRequest request = DescribeVolumesRequest.builder()
-				.volumeIds(attachmentId)
-				.build();
-
-		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
-		DescribeVolumesResponse volumesResponse = client.describeVolumes(request);
-
-		AttachmentInstance attachmentInstance = mountAttachmentInstance(volumesResponse);
-		return attachmentInstance;
-	}
+    @Override
+    public void deleteInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, attachmentOrder.getInstanceId()));
+
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        String volumeId = attachmentOrder.getVolumeId();
+
+        DetachVolumeRequest request = DetachVolumeRequest.builder()
+                .volumeId(volumeId)
+                .build();
+
+        try {
+            client.detachVolume(request);
+        } catch (Exception e) {
+            LOGGER.error(String.format(Messages.Error.ERROR_WHILE_REMOVING_RESOURCE, RESOURCE_NAME, volumeId), e);
+            throw new UnexpectedException();
+        }
+    }
+
+    @Override
+    public AttachmentInstance getInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, attachmentOrder.getInstanceId()));
+
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        String attachmentId = attachmentOrder.getInstanceId();
+        
+        DescribeVolumesRequest request = DescribeVolumesRequest.builder()
+                .volumeIds(attachmentId)
+                .build();
+        
+        DescribeVolumesResponse response = AwsV2CloudUtil.doDescribeVolumesRequest(client, request);
+        return mountAttachmentInstance(response);
```

Comment:
`mount` is a reserved word for the storage/filesystem stuff. I think it is createAttachmentInstance, right? or build ...

Reply:
nice